### ISSUE TITLE
column-resizing option slow down the grid 

### ISSUE THREAD

**Issue by jibap** [angular-ui/ng-grid#3623]
Hi, 

I'm in v3.0.0-rc.21 - 2015-04-28

When i add the column-resizing option, the grid slow down (filtering, hiding columns...)
I made a plunker : http://plnkr.co/edit/9xVHl1UTATrbQYlO94JW?p=preview based on 404_large_data_sets_and_performance tutorial plunker

More there is columns, more the grid is slow and if i remove the option, i have no slowing down...

Maybe i am not using correctly ?

Thanks

**Comment by c0bra** [comment]
Filtering and hiding columns both run in about the same time when I have the `ui-grid-resize-columns` directive both on the grid and off. I cannot notice any perceptibly performance difference.

Are you seeing this in particular browser?

**Comment by jibap** [comment]
On the plunker, If i remove the <code>ui-grid-resize-columns</code> directive, the execution time of search is devided by 2. I tested on Iron (Version 39.0.2100.0), Firefox(38.0.1) and IE 11.019, the difference is more perceptible with chrome.

**Comment by c0bra** [comment]
What are your machine specs? Have you tried on another machine?  I honestly cannot see any difference at all...

**Comment by jibap** [comment]
![image](https://cloud.githubusercontent.com/assets/3915029/7954409/ab2c2656-09cf-11e5-9aa8-df0738bb0930.png)
I tested on other machine with the same effect

**Comment by c0bra** [comment]
Strange that should be more than enough.

- [ ] I'll make sure to test your plunker on my home machine, which is pretty old.

I hate to keep asking you for stuff, but would you be able to do a quick screencap of you using the plunker that shows the slow-down?

**Comment by jibap** [comment]
OK, 
first <b>with</b> directive : 
![avec](https://cloud.githubusercontent.com/assets/3915029/8277765/4a81ab72-18c6-11e5-80bb-9e612fbbc2c5.gif)

and <b>whitout</b> : 
![sans](https://cloud.githubusercontent.com/assets/3915029/8277768/502b9358-18c6-11e5-80d5-d6bdfe62b55a.gif)

when i cut the capture on video editor, i get precisely time frames and the version with directive take 2 times longer ( 300 ms => 700ms)

Thanks for your interest

**Comment by c0bra** [comment]
OK, interesting. It's hard to tell with the loop but it looks like maybe you're entering the filter twice?  One the second screencap it looks like the filter takes a bit at first, and then the second time it filters it's almost instantaneous. Is that right, or am I understand the clips wrong?

**Comment by jibap** [comment]
Yes, you're right, i observed that the grid is faster responsive after some manipulations on it (filter, sort...) as if need to wake up

**Comment by lebolo** [comment]
I'm also seeing this problem. I noticed that the filtering isn't being processed until *after* some type of column-resize processing completes.

For example, if you look at the HTML around the filter you're typing into, you'll see a lot of blocks being generated that look like

```html
<!-- ngIf: grid.options.enableColumnResizing -->
<!-- end ngIf: grid.options.enableColumnResizing -->
```

Once these blocks stop being generated, *then* the filter gets processed.

![badfilter](https://cloud.githubusercontent.com/assets/2286572/8729162/71a7c4ee-2bb7-11e5-8d7a-5575830016db.gif)

**Comment by lebolo** [comment]
Bumping this to see if anyone has an idea of why this is happening (or can point me in the right direction to solve it). I'm still seeing this effect in 3.0.7.

**Comment by itzik25b** [comment]
Hi,
I see that the timeout causing this issue since it goes over all columns on each typing
(line 21891):
var waitDisplay = function(){
                $timeout(displayResizers);
              };

what is that timeout for?

**Comment by prettycode** [comment]
+1 Definitely observing this still with 3.1.1. Adding `ui-grid-resize-columns` slows down the column filters (i.e. `enableFiltering: true`) by about 3 - 4x with the dataset I'm using.

**Comment by lebolo** [comment]
We should also remove the `waiting-response` label from this issue, since that may be affecting the priority of this issue =P

**Comment by hoonto** [comment]
+1 seeing this problem in 3.1.1 also, haven't looked further however yet to see what is happening

**Comment by JonRichter** [comment]
we had to disable the ui-grid-resize-columns option as well and our sluggish key stroke issues went away. Hopefully we can figure this out so we can get that valuable resize option back at some point. We were seeing significant performance issues, you literally were typing a keystroke and waiting.

**Comment by DPalkar** [comment]
We also see the same issue, with ui-grid-resize-columns directive with 3.1.1.

**Comment by k-suzuki** [comment]
Are there any plans to resolve this problem?
In my environment, this problem is occurring.
Please let me know if there is a countermeasure other than setting to enableFiltering: False.

**Comment by rpgehlot** [comment]
I am using v4.0.10 and this problem is still there. Can anybody from core team look at this issue. I use grid to update data per sec(trading data like pos,pnl etc.) and filtering symbols takes a lot of time for dataset of size 1k. Only option i can see is to disable **enableColumnResizing** but this is a very much required feature.
